File: scripts/extract_clients.js
#!/usr/bin/env node
/**
 * Extract client names from Excel file and generate SQL migration
 */

import fs from 'fs';
import path from 'path';
import { fileURLToPath } from 'url';
import xlsx from 'xlsx';

const __dirname = path.dirname(fileURLToPath(import.meta.url));
const projectRoot = path.join(__dirname, '..');

async function extractClientsFromExcel() {
  try {
    // Find the Excel file
    const files = fs.readdirSync(projectRoot);
    let excelFile = files.find(f => f.includes('client') && f.endsWith('.xls'));
    
    if (!excelFile) {
      console.error('[v0] Excel file not found in project root');
      process.exit(1);
    }

    const excelPath = path.join(projectRoot, excelFile);
    console.log(`[v0] Reading Excel file: ${excelPath}`);

    // Read the Excel file
    const workbook = xlsx.readFile(excelPath);
    const sheetName = workbook.SheetNames[0];
    const worksheet = workbook.Sheets[sheetName];
    const data = xlsx.utils.sheet_to_json(worksheet);

    console.log(`[v0] Found ${data.length} rows`);

    // Extract client information
    const clients = [];
    const seenNames = new Set();

    data.forEach((row, index) => {
      // Look for columns that might contain ID and name
      let id = null;
      let name = null;

      // Try different column names
      for (const key of Object.keys(row)) {
        const value = String(row[key]).trim();
        if (!value || value === 'ID' || value === 'Name' || value === 'NAMES') continue;

        // First column usually ID
        if (!id && /^\d+$/.test(value)) {
          id = value;
        }
        // Other columns might be name
        if (!name && value.length > 2 && /^[a-zA-Z\s,.-]+$/.test(value)) {
          name = value;
        }
      }

      if (name && !seenNames.has(name)) {
        clients.push({
          id: id || clients.length + 1,
          name: name
        });
        seenNames.add(name);
      }
    });

    console.log(`[v0] Extracted ${clients.length} unique clients`);
    console.log(JSON.stringify(clients, null, 2));

    return clients;
  } catch (error) {
    console.error('[v0] Error:', error.message);
    process.exit(1);
  }
}

extractClientsFromExcel();

File: scripts/extract_clients.py
#!/usr/bin/env python3
"""
Extract client names from Excel file and generate SQL insert statements
"""
import xlrd
import sys
import json

def extract_clients_from_excel(excel_file):
    """Extract client names from Excel file"""
    try:
        # Load the Excel workbook using xlrd for .xls format
        wb = xlrd.open_workbook(excel_file)
        ws = wb.sheet_by_index(0)
        
        clients = []
        
        # Iterate through rows, skip header (row 0)
        for row_idx in range(1, ws.nrows):
            # Get values from columns A (0) and B (1)
            client_id = str(ws.cell_value(row_idx, 0)).strip()
            client_name = str(ws.cell_value(row_idx, 1)).strip() if ws.ncols > 1 else None
            
            # Skip empty rows and header-like rows
            if client_name and client_name.upper() not in ['NAME OF CLIENT', '', 'NAMES']:
                clients.append({
                    'id': client_id,
                    'name': client_name
                })
        
        return clients
    except Exception as e:
        print(f"Error extracting clients: {e}", file=sys.stderr)
        return []

if __name__ == '__main__':
    import os
    import glob
    
    # Find the Excel file
    excel_files = glob.glob('/**/*clientlistshort*.xls', recursive=True)
    
    if excel_files:
        excel_path = excel_files[0]
    else:
        excel_path = '9044 clientlistshort031826135150350.xls'
    
    clients = extract_clients_from_excel(excel_path)
    
    if not clients:
        print("No clients found or error reading file", file=sys.stderr)
        sys.exit(1)
    
    # Output as JSON for easy parsing
    print(json.dumps(clients, indent=2))
    print(f"\n# Total clients: {len(clients)}", file=sys.stderr)
